.github/workflows/main.yml

```diff
@@ -59,19 +60,19 @@ jobs:
       ############################################################
 
       - name: Compile
-        run: $SBT compile test:compile it:compile
+        run: $SBTNOCOV compile test:compile it:compile
```

Comment:
I think we need to be careful how we do no coverage to prevent unnecessary compilation. For example, if you do the folowoing:
```
sbt compile
sbt coverage test
```
then I *think* that the first will compile everything. The second will then have to recompile everything with coverage enabled and then run the test. So we end up compiling everything twice. That was the reason why "coverage" was added to the SBT command so that we dont' do a bunch of unnessary compilation. If we switch back and forth between cov and nocov, we might end up recompiling things a bunch.

Maybe we should jus do all things that don't do any kindof publishing first with coverage enabled, then at then do all the things that do publishing and require no coverage at the en. Then we at least only duplicate compilation once.

This is assuming that there'es a noticable different in time it takes to finish all the github actions. If it still takes about thes same amount of time, it might not be worth the effort.

Reply:
So the compile step still uses the coverage option. It's just the doc, and packaging steps that use the no-coverage option.

Reply:
Ah you're right, looks like you do all the nocov stuff first. This should be fine.

Reply:
No I'm not sure. The basic compile step should be SBT with coverage I believe. Only the doc, package steps should use SBTNOCOV.

Reply:
Yeah, maybe that's more correct. Maybe just move the doc and package steps to the end with nocov?

Reply:
I think moving it to the end might cause issues for the publishing of the coverage results. That's one of the last things to run because it wants to ensure everything passes before we get to that step, correct?

Reply:
Let's leave it the way it is now. It does all build steps, including packaging and doc gen, then it does all testing. I think that is sensible given how time consuming the testing is.